plugins/modules/nd_compliance_requirement_config_template.py

```diff
@@ -0,0 +1,202 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: nd_compliance_requirement_config_template
+version_added: "0.2.1"
+short_description: Manage template configuration type compliance requirements
+description:
+- Manage template configuration type on Cisco Nexus Dashboard Insights (NDI).
+author:
+- Akini Ross (@akinross)
+options:
+  insights_group:
+    description:
+    - The name of the insights group.
+    type: str
+    required: yes
+    aliases: [ fab_name, ig_name ]
+  name:
+    description:
+    - The name of the compliance requirement.
+    type: str
+  description:
+    description:
+    - The description of the compliance requirement.
+    type: str
+    aliases: [ descr ]
+  enabled:
+    description:
+    - Enable the compliance requirement.
+    type: bool
+  sites:
+    description:
+    - Names of the Assurance Entities.
+    type: list
+    elements: str
+  file:
+    description:
+    - Name of the template file to upload.
+    type: str
+  selector_based_on_tags:
+    description:
+    - Enable object selection based on Tag Annotation or Tag Instance.
+    type: bool
+    default: false
+  state:
+    description:
+    - Use C(query) for retrieving the version object.
+    type: str
+    choices: [ query, absent, present ]
+    default: query
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Get all compliance template configuration type requirements
+  cisco.nd.nd_compliance_requirement_config_template:
+    insights_group: igName
+    state: query
+  register: query_results
+- name: Get a specific compliance template configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_template:
+    insights_group: igName
+    name: complianceRequirementName
+    state: query
+  register: query_results
+- name: Create compliance template configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_template:
+    insights_group: igName
+    name: complianceRequirementName
+    sites:
+    - siteName1
+    - siteName2
+    enabled: false
+    file: fileName.json
+    state: present
+- name: Delete compliance template configuration type requirement
+  cisco.nd.nd_compliance_requirement_config_template:
+    insights_group: igName
+    name: complianceRequirementName
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize_dict
+from ansible_collections.cisco.nd.plugins.module_utils.ndi import NDI
+from ansible_collections.cisco.nd.plugins.module_utils.ndi_argument_specs import compliance_base_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(compliance_base_spec())
```

Comment:
Should we use a documentation fragment for this compliance_base_spec part?

Reply:
yes good idea will do